Kernel/Devices/Storage/USB/BulkSCSIInterface.cpp

```diff
@@ -5,215 +5,37 @@
  */
 
 #include <Kernel/Bus/USB/USBController.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/Storage/StorageManagement.h>
 #include <Kernel/Devices/Storage/USB/BulkSCSIInterface.h>
 #include <Kernel/Devices/Storage/USB/SCSIComands.h>
 
 namespace Kernel::USB {
 
-BulkSCSIInterface::BulkSCSIInterface(LUNAddress logical_unit_number_address, u32 hardware_relative_controller_id, size_t sector_size, u64 max_addressable_block, USB::Device& device, NonnullOwnPtr<BulkInPipe> in_pipe, NonnullOwnPtr<BulkOutPipe> out_pipe)
-    : StorageDevice(logical_unit_number_address, hardware_relative_controller_id, sector_size, max_addressable_block)
-    , m_device(device)
+BulkSCSIInterface::BulkSCSIInterface(StorageDevice::LUNAddress logical_unit_number_address, size_t sector_size, u64 max_addressable_block, USB::Device& device, NonnullOwnPtr<BulkInPipe> in_pipe, NonnullOwnPtr<BulkOutPipe> out_pipe)
+    : m_device(device)
     , m_in_pipe(move(in_pipe))
     , m_out_pipe(move(out_pipe))
 {
-    // Note: If this fails, it only means that we may be inefficient in our way
-    //       of talking to the device.
-    (void)query_characteristics();
+    auto storage_device = DeviceManagement::try_create_device<BulkSCSIStorageDevice>(
+        *this,
+        *m_out_pipe,
+        *m_in_pipe,
+        logical_unit_number_address,
+        device.address(), // FIXME: Figure out a better ID to put here
+        sector_size,
+        max_addressable_block)
+                              .release_value_but_fixme_should_propagate_errors();
```

Comment:
Should we make the caller create this and pass it in? Otherwise, have a ::create() function. Constructors shouldn't do things that can fail.

Reply:
This is just a temporary fixme for that commit. The next commit (e05c54adb53ba0f63c0e8ae4bc02ef61b1461104) adds a fallible ::initialize function, which replaces it with a TRY.